scripts/bazzite-setup.sh

```
@@ -245,18 +283,29 @@ systemctl --user restart padctl.service 2>/dev/null || true
 #         their IPC socket under $XDG_RUNTIME_DIR (/run/user/<uid>/padctl.sock),
 #         and the CLI's default resolver finds it correctly from the user shell.
 #         The daemon needs a few seconds post-restart to run device init + bind
-#         its IPC socket, so retry a few times before giving up. ---
+#         its IPC socket, so retry a few times before giving up. "no-devices"
+#         is a permanent error for this session (no retry will help) — the
+#         binding in /etc/padctl/config.toml still takes effect on the next
+#         hot-plug, so we short-circuit with a clearer message. ---
 if [[ -n "$MAPPING" ]]; then
     mapping_applied=false
+    no_devices=false
     for attempt in 1 2 3 4 5 6; do
```

Shellcheck (0.11.0)

[warning] attempt appears unused. Verify use (or export if used externally).

(SC2034)